arroyo/processing/strategies/dead_letter_queue/dead_letter_queue.py

```diff
@@ -33,17 +37,23 @@ def __init__(
     def poll(self) -> None:
         try:
             self.__next_step.poll()
-        except InvalidMessages as e:
-            self.__metrics.increment("dlq.received_message", len(e.messages))
-            self.__policy.handle_invalid_messages(e)
+        except (InvalidMessage, InvalidBatchedMessages) as e:
+            self._handle_invalid_messages(e)
 
     def submit(self, message: Message[TPayload]) -> None:
         assert not self.__closed
         try:
             self.__next_step.submit(message)
-        except InvalidMessages as e:
-            self.__metrics.increment("dlq.received_message", len(e.messages))
-            self.__policy.handle_invalid_messages(e)
+        except (InvalidMessage, InvalidBatchedMessages) as e:
+            self._handle_invalid_messages(e)
+
+    def _handle_invalid_messages(
+        self, e: Union[InvalidMessage, InvalidBatchedMessages]
+    ) -> None:
+        if isinstance(e, InvalidMessage):
+            e = InvalidBatchedMessages([e])
+        self.__metrics.increment(RECEIVED_MESSAGE_METRIC, len(e.exceptions))
+        self.__policy.handle_invalid_messages(e)
```

Comment:
With this implementation, a policy decides how to deal with multiple bad messages. If I had just looped and submitted bad messages one by one to a policy, we would lose part of the batch of bad messages if the policy decides to raise an exception part way through.

This would make our count policy have an inaccurate count since it would not have counted up all bad messages before throwing the exception.